compiler/rustc_query_system/src/ich/impls_syntax.rs

```diff
@@ -75,7 +75,7 @@ impl<'a> HashStable<StableHashingContext<'a>> for SourceFile {
             ref normalized_pos,
         } = *self;
 
-        (name_hash as u64).hash_stable(hcx, hasher);
+        (u128::from_le_bytes(name_hash) as u64).hash_stable(hcx, hasher);
```

Comment:
```suggestion
        name_hash.hash_stable(hcx, hasher);
```
?

Reply:
I don't understand why this was deliberately truncated in the first place. Such truncation is done in multiple places, sometimes with a comment trying to justify it for performance. On account of the huge amount of code I need to change to add wrapper types for hashes, I'm leaving the truncations in place for now.